hs_core/views/resource_folder_hierarchy.py

```diff
@@ -42,6 +42,7 @@ def data_store_structure(request):
     store_path = str(store_path).strip()
     if not store_path:
         return HttpResponse('Bad request - store_path cannot be empty', status=400)
+    # this is federated if warranted.

```

Comment:
I am deferring the refactoring of the other iRODS code because I plan to move a lot of it (in fact, almost all of it) inside ResourceFile. This will de-duplicate everything.